File: elixir/isbn-verifier/lib/isbn_verifier.ex
defmodule IsbnVerifier do
  @doc """
    Checks if a string is a valid ISBN-10 identifier

    ## Examples

      iex> IsbnVerifier.isbn?("3-598-21507-X")
      true

      iex> IsbnVerifier.isbn?("3-598-2K507-0")
      false

  """
  @spec isbn?(String.t()) :: boolean
  def isbn?(isbn) do
    cleaned = String.replace(isbn, "-", "")
    if Regex.match?(~r/^\d{9}[\dX]$/, cleaned) do
      checksum = 
        cleaned
        |> String.graphemes()
        |> Enum.map(& if &1 == "X", do: 10, else: String.to_integer(&1))
        |> Enum.with_index(fn value, index -> value * (10 - index) end)
        |> Enum.sum()
      rem(checksum, 11) == 0
    else
      :false
    end
  end
end

File: python/isbn-verifier/isbn_verifier.py
def is_valid(isbn):
    isbn = isbn.replace("-", "")

    if len(isbn) != 10:
        return False

    if not isbn[:9].isdigit():
        return False

    if not (isbn[-1].isdigit() or isbn[-1] == "X"):
        return False

    checksum = sum(
        (10 - index) * (10 if char == "X" else int(char))
        for index, char in enumerate(isbn)
    )

    return checksum % 11 == 0
